metrics/stackdriver_exporter.go

```diff
@@ -40,8 +40,10 @@ const (
 	// defaultCustomMetricSubDomain is the default subdomain to use for unsupported metrics by monitored resource types.
 	// See: https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#MetricDescriptor
 	defaultCustomMetricSubDomain = "knative.dev"
-	// secretNamespaceDefault is the namespace to search for a k8s Secret to pass to Stackdriver client to authenticate with Stackdriver.
-	secretNamespaceDefault = "default"
+	// DefaultSecretNamespace is the default namespace to search for a k8s Secret to pass to Stackdriver client to authenticate with Stackdriver.
+	DefaultSecretNamespace = "default"
```

Comment:
I'm not sure that "default" is a good option here, but putting this in `pkg` makes it hard to default to something like "knative-serving".

I'm not sure if there's an issue filed to figure out the best setting here and come back to this setting later.

Reply:
I think leaving DefaultSecretNamespace as "default" because that should really be the default for any namespace value that can't be predetermined. 

If the consuming package doesn't overwrite the namespace, the pod will just attempt to read a Secret out of the "default" namespace, which will get rejected if the pod doesn't have access to read Secrets anyway.

Reply:
Also, we cannot allow reading an env variable because that is user-configurable which will be a security issue for knative/serving when it starts to propagate the Secret specified here to queue-proxy.